R/viz.R

```diff
@@ -2918,84 +2918,105 @@ fancy_svbar = function(this_sample,
+#' chr1_cns = fancy_cnbar(this_sample = "HTMCP-01-06-00422-01A-01D", chr_select = c(1))
+#' cns = fancy_cnbar(this_sample = "HTMCP-01-06-00422-01A-01D")
 #'
-fancy_cnlohbar = function(this_sample,
-                          plot_subtitle = "CNV states and LOH Regions For Selected Contigs",
-                          chr_select = paste0("chr", c(1:22)),
-                          coding_only = FALSE,
-                          from_flatfile = FALSE,
-                          use_augmented_maf = FALSE){
+fancy_cnbar = function(this_sample,
+                       plot_subtitle = "n CNV Segments (barplots, left y-axis), n Affected bases for each CN state",
+                       chr_select = paste0("chr", c(1:22)),
+                       include_cn2 = TRUE) {
 
   #get maf data for a specific sample.
-  maf = assign_cn_to_ssm(this_sample = this_sample, coding_only = coding_only, from_flatfile = from_flatfile, use_augmented_maf = use_augmented_maf)$maf
+  maf = get_sample_cn_segments(this_sample = this_sample, multiple_samples = FALSE, streamlined = FALSE, from_flatfile = TRUE)
 
   #add chr prefix if missing
-  if(!str_detect(maf$Chromosome, "chr")[5]){
-    maf = mutate(maf, Chromosome = paste0("chr", Chromosome))
+  if(!str_detect(maf$chrom, "chr")[2]){
+    maf = mutate(maf, chrom = paste0("chr", chrom))
   }
 
   #read maf into R and select relevant variables and transformt to factor.
-  maf_df = dplyr::select(maf, Chromosome, Start_Position, End_Position, Variant_Type, LOH, CN) %>%
-    mutate_at(vars(Chromosome, Variant_Type), list(factor))
+  maf_df = dplyr::select(maf, chrom, start, end, CN) %>%
+    mutate_at(vars(chrom), list(factor))
 
   #subsetting maf based on user-defined parameters
-  maf_df = maf_df[maf_df$Chromosome %in% chr_select, ]
+  maf_df = maf_df[maf_df$chrom %in% chr_select, ]
 
   #transform data type
-  maf_df$LOH = as.factor(maf_df$LOH)
   maf_df$CN = as.factor(maf_df$CN)
 
   #count levels of factor
-  loh_count = dplyr::filter(maf_df, LOH == 1) %>%
-    group_by(LOH) %>%
-    summarize(count = n())
+  if(include_cn2){
+    cn_states = c(0:15)
+    cns_count = dplyr::filter(maf_df, CN %in% cn_states) %>%
+      group_by(CN) %>%
+      summarize(count = n())
+  }else{
+    cn_states = c(0:1, 3:15)
+    cns_count = dplyr::filter(maf_df, CN %in% cn_states) %>%
+      group_by(CN) %>%
+      summarize(count = n())
+  }
 
-  loh_count$Type = paste0("LOH-", loh_count$LOH)
-  loh_count = dplyr::select(loh_count, count, Type)
+  cns_count$Type = paste0(cns_count$CN)
+  cns_count = dplyr::select(cns_count, count, Type)
+  cns_count = dplyr::filter(cns_count, Type != 0)
 
-  cns_count = dplyr::filter(maf_df, CN == 1 | CN == 3 | CN == 4) %>%
-    group_by(CN) %>%
-    summarize(count = n())
+  #compute lenght of cn segments and transform for plotting
+  l_cn_seg = maf
+  l_cn_seg$lenght = l_cn_seg$end - l_cn_seg$start
+  l_cn_seg$CN = as.factor(l_cn_seg$CN)
+  l_cn_seg = dplyr::filter(l_cn_seg, CN != 0)
```

Comment:
Another good catch, when I developed these functions I wrongly assumed that CN states 0 and 1 always would be interpreted as LOH events. I've made updates to this and the plots (i.e ideograms) now show CN states 0 and 1. So, this line needs to be updated so that CN0 is not removed.